src/zbarimg.ts

```diff
@@ -1,22 +1,28 @@
 import execFile from './exec'
 
+interface ZBarImageParams {
+  readonly filepath: string
+  readonly encoding?: BufferEncoding
+}
+
 export interface ZBarImage {
-  readQRCodeFromImage(filepath: string): Promise<Buffer | undefined>
+  readQRCodeFromImage(params: ZBarImageParams): Promise<Buffer | undefined>
 }
 
 /**
  * Expects our hacked-up version of zbarimg which returns a hex string
  * when it's a binary QR Code
  */
 export class RealZBarImage implements ZBarImage {
-  public async readQRCodeFromImage(
-    filepath: string
-  ): Promise<Buffer | undefined> {
+  public async readQRCodeFromImage({
+    filepath,
+    encoding = 'hex',
```

Comment:
A lot of nodejs APIs pass encoding as a separate argument, so that might be less surprising. Still, won’t the output always be hex if we’re passing `--raw`?

Reply:
No, raw just means no processing, hex is only for the binary encoding back we built

Reply:
Also raw means no headers, just raw data.

Reply:
wondering if I should instead just return the bytes and let the calling code figure it out.

Reply:
So, just to clarify, zbar decides whether it outputs hex or ascii based on the QR code?

Reply:
well, I'm confused. I know we hacked it to make it return hex, and it looks like it's doing it *for all* cases, whether it's a URL or not, whether we specify `--raw` or not. So ultimately, we don't strictly need this feature here. I'm going to pull it back to just turning the single argument into an object for future extensibility and call it a day :)

Reply:
@eventualbuddha ready for your re-review, down-scoped to just future-proofing the API call.